scripts/cluster_buildings.py

```diff
@@ -160,19 +152,26 @@ def get_central_points_geojson_with_buildings(
 
     configure_logging(snakemake)
 
-    extract_points(
-        snakemake.input["buildings_geojson"],
-        snakemake.output["cleaned_buildings_geojson"],
-    )
+    crs = snakemake.params.crs["area_crs"]
+    house_area_limit = snakemake.params.house_area_limit["area_limit"]
 
     get_central_points_geojson(
-        snakemake.output["cleaned_buildings_geojson"],
+        snakemake.input["buildings_geojson"],
         snakemake.output["clusters"],
         snakemake.config["buildings"]["n_clusters"],
+        crs,
+        house_area_limit,
     )
 
     get_central_points_geojson_with_buildings(
-        snakemake.output["cleaned_buildings_geojson"],
+        snakemake.input["buildings_geojson"],
         snakemake.output["clusters_with_buildings"],
         snakemake.config["buildings"]["n_clusters"],
+        crs,
+        house_area_limit,
+    )
+
+    get_number_type_buildings(
+        snakemake.output["clusters_with_buildings"],
+        snakemake.output["number_buildings_type"],
```

Comment:
In the functions I see a lot of input/output. Ideally `snakemake.input["buildings_geojson"]` can be loaded outside and passed to the function so avoiding reading it multiple times.
It saves time.
Same applies for the others

Reply:
Perfect!

Reply:
In case it is okay to compact the functions and keep the three outputs, the buildings_geojson would only be called once, can I leave it in instead of putting it out?